draft-ietf-tls-extended-key-update.md

```diff
@@ -806,6 +809,19 @@ the application might retain the previous exporter secret until its
 replay window no longer accepts packets protected with keys derived from that
 secret, as described in Section 3.3.2 of {{!RFC3711}}.
 
+Applications may receive data protected under a newly derived exporter secret
+before notification is delivered. In such cases, if decryption using the
+previous exporter secret fails or if the application protocol provides
+an explicit indication of new keying material (for example, through a
+key identifier or epoch field), the application should explicitly request
+the corresponding exporter-derived keying material.
+
+The existing exporter interface defined in {{Section 7.5 of TLS}} remains unchanged
+and continues to operate as specified. When Extended Key Update is used, this interface
+produces keying material for the currently active epoch. Implementations may also
+provide an additional exporter interface that accepts an explicit epoch parameter,
+allowing applications to request keying material for a specific epoch.
```

Comment:
This is not acceptable (massive breaking change to TLS). The exporter interface defined in Section 7.5 of TLS **MUST NOT** update its epoch during connection lifetime. Using current epoch would totally break most applications using exporters.

The "additional exporter interface" is the only way way to export keying material that is updated during connection lifetime. Furthermore, that interface does not have any implicit epoch behavior either, taking the epoch to use as parameter.

Reply:
Fixed text